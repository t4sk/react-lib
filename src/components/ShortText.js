import React from "react"
import PropTypes from "prop-types"

export function ShortText(props) {
  const { width, text } = props

  return (
    <div
      style={{
        width: width,
        overflow: "hidden",
        textOverflow: "ellipsis",
        whiteSpace: "nowrap",
      }}
    >
      {text}
    </div>
  )
}

ShortText.propTypes = {
  width: PropTypes.number.isRequired,
  text: PropTypes.node.isRequired,
}

ShortText.defaultProps = {
  width: 100,
}

export default ShortText
