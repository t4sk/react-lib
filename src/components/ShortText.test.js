import React from "react"
import { shallow } from "enzyme"
import { ShortText } from "./ShortText"

const props = {
  text: "0x000000462a632b33612C857e2567B3a73aBf17Cb",
}

test("it renders", () => {
  const component = shallow(<ShortText {...props} />)

  expect(component).toMatchSnapshot()
})
