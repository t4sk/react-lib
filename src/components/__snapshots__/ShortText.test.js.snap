// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<div
  style={
    Object {
      "overflow": "hidden",
      "textOverflow": "ellipsis",
      "whiteSpace": "nowrap",
      "width": 100,
    }
  }
>
  0x000000462a632b33612C857e2567B3a73aBf17Cb
</div>
`;
